### PR TITLE
Kinetis:Add compatible gpiosetevent interface

### PR DESCRIPTION
**Pull request by davids5**
## Summary
   This PR adds a compatible gpiosetevent interface. The helper call the low level functions while providing a 
consistent API with `xxxx_ gpiosetevent(uint32_t pinset, bool risingedge, bool fallingedge, bool event, xcpt_t func, void *arg);`

## Impact
None. It adds the new API but has not impact on existing code.

## Testing

NXP_RDDRONE_K66 in PX4.

### CODE REVIEW

File: arch/arm/src/kinetis/kinetis_pinirq.c [hunk 1]
```diff
@@ -459,4 +475,56 @@ void kinetis_pinirqdisable(uint32_t pinset)
...
+ *  - func:   when non-NULL, generate interrupt
+ *  - arg:    Argument passed to the interrupt callback
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure indicating the
+ *   nature of the failure.
+ *
+ ****************************************************************************/
+
+int kinetis_gpiosetevent(uint32_t pinset, bool risingedge, bool fallingedge,
```
Comment (xiaoxiang781216) [inline arch/arm/src/kinetis/kinetis_pinirq.c:498]: But I can't see any place use this new function.

Comment (patacongo) [reply arch/arm/src/kinetis/kinetis_pinirq.c:498]: Looks like a pure, useless wrapper function that only serves the PX4 project.  I have resisted all such changes in the past.

Even thought the function is not used, all builds will pay the overhead of the PX4-specific function.

Comment (davids5) [reply arch/arm/src/kinetis/kinetis_pinirq.c:498]: Yes the board is will be used on has not been up streamed yet. 

If the linker is set up correctly dead code will be removed.

Comment (patacongo) [reply arch/arm/src/kinetis/kinetis_pinirq.c:498]: > If the linker is set up correctly dead code will be removed.

Why not put the function in a separate file?

**Pull request closed without merging**